app/src/main/java/io/apicurio/registry/services/auth/CustomAuthenticationMechanism.java

```diff
@@ -101,4 +183,11 @@
     public HttpCredentialTransport getCredentialTransport() {
         return new HttpCredentialTransport(HttpCredentialTransport.Type.AUTHORIZATION, "bearer");
     }
+
+    private Uni<SecurityIdentity> authenticateWithClientCredentials(Pair<String, String> clientCredentials, RoutingContext context, IdentityProviderManager identityProviderManager) {
+        OidcAuth oidcAuth = new OidcAuth(httpClient, clientCredentials.getLeft(), clientCredentials.getRight());
+        final String jwtToken = oidcAuth.authenticate();//If we manage to get a token from basic credentials, try to authenticate it using the fetched token using the identity provider manager
```

Comment:
No need to close this now because the lifecycle of the client is external to `OidcAuth` right?

Reply:
Yes, exactly.